### PR TITLE
DSD-546: Adding a11y documentation

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-546](https://jira.nypl.org/browse/DSD-546)

## This PR does the following:
- Adds general information and documentation about accessibility in the README
- Fixes minor a11y issues in Storybook

## How has this been tested?

Storybook.

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr794-if8w6nlhvixmn5cq85iq4m0z8tfgeuig.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61ae8c0c821ec8a4b60a83e3

File: src/styles/base/_place-holder.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .placeholder {
-  --color-place-holder: var(--nypl-colors-ui-gray-light-cool);
+  --color-place-holder: var(--nypl-colors-ui-gray-dark);
```
Comment (bigfishdesign13) [inline src/styles/base/_place-holder.scss:2]: Was this change made for `a11y` reasons?  `gray.dark` is the lightest gray that is compliant on top of white?

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

File: src/styles/base/_place-holder.scss (hunk 1 above)
Comment (EdwinGuzman) [reply src/styles/base/_place-holder.scss:2]: Yes, it seems like it is the most compliant one _for that font size_. A color matrix would be nice to have (maybe similar to what we had before https://nypl.github.io/design-toolkit/sections/color-a11y.html).

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr794-if8w6nlhvixmn5cq85iq4m0z8tfgeuig.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61ae8c0c821ec8a4b60a83e3

**Pull request merged**